lib/core/asdf/default/90_asdf.sh

```diff
@@ -3,18 +3,23 @@
 
 # This adds the asdf shims to the shell.
 
-SHELL=$(ps -cp "$$" -o command="")
+function source_asdf() {
+  local CURRENT_SHELL
+  CURRENT_SHELL=$(ps -cp "$$" -o command="")
```

Comment:
Kinda wonder if you need this variable instead of just using `$SHELL`.

Reply:
This gets sourced during shell startup and I've observed `$SHELL` being `/bin/sh` during that. Maybe it's a MacOS thing.